### PR TITLE
feat(discover2): Add columns in-between columns

### PR DESCRIPTION
**Pull request by dashed**
This PR is a continuation of https://github.com/getsentry/sentry/pull/14980

## TODO

- [x] add columns in-between columns (only if there are at least 2 columns)
- [ ] ~add column to the left.~ EDIT: nah.
- [x] ~ping either Dora or John for some designs.~ EDIT: low priority
- [x] merge https://github.com/getsentry/sentry/pull/14980

-------

Closes SEN-1172

### CODE REVIEW

File: src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+type Props = {
+  onClick: () => void;
+};
+
+const AddColumnButton = (props: Props) => {
+  const {onClick} = props;
+  return (
+    <Wrapper onClick={onClick}>
+      <ToolTip title={t('Add Column')}>
+        <div style={{width: `${BUTTON_SIZE}px`, height: `${BUTTON_SIZE}px`}}>
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx]: Can we use a styled component here?

File: src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          <Vertical />
+          <Horizontal />
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx]: We should get an SVG icon for this.

File: src/sentry/static/sentry/app/components/gridEditable/gridHeadCell.tsx [hunk 3]
```diff
@@ -118,27 +122,54 @@ class GridHeadCell<Column extends GridColumnHeader> extends React.Component<
...
+      isColumnDragging,
+    } = this.props;
+
+    if (isLast || !isEditing || isColumnDragging) {
+      return null;
+    }
+
+    return (
+      <React.Fragment>
+        {FLAG_GRID_RESIZABLE && <GridHeadCellResizer isEditing={isEditing} />}
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/gridEditable/gridHeadCell.tsx]: Will these two components end up in the same small space?

Comment (dashed) [reply src/sentry/static/sentry/app/components/gridEditable/gridHeadCell.tsx]: I separated them.

File: src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx (hunk 1 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx]: Yes we can.

File: src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx (hunk 2 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx]: I'll try asking Dora and John.

Comment (markstory) [reply src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx]: Or re-use the + in a circle we already have.

Comment (dashed) [reply src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx]: Ok I'll look into it.

**Review by markstory — APPROVED** [review]
Looks good to me once the icon is sorted out.

File: src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx (hunk 2 above)
Comment (dashed) [reply src/sentry/static/sentry/app/components/gridEditable/addColumnButton.tsx]: Done.

**Pull request merged**